src/macos/mod.rs

```diff
@@ -60,8 +60,8 @@ impl Verifier {
         }
     }
 
-    pub fn verify(&self) -> Result<Context, Error> {
-        self.check_validity("anchor trusted")?; // This is the most generic verification
+    pub fn verify(&self, requirement: &str) -> Result<Context, Error> {
```

Comment:
Should `requirement` be an enum, instead of requiring the caller to know the right magic string to pass in?

Reply:
We can have an enum, sure, but I suspect in the future our customers might want tighter verifications.

For e.g.,: we can extend it to include:

```
identifier "com.google.santa" and anchor apple generic and certificate 1[field.1.2.840.113635.100.6.2.6] /* exists */ and certificate leaf[field.1.2.840.113635.100.6.1.13] /* exists */ and certificate leaf[subject.OU] = EQHXZ8M8AV
```

to further tighten the check.

```
codesign --display -r -  /Applications/Santa.app  
designated => identifier "com.google.santa" and anchor apple generic and certificate 1[field.1.2.840.113635.100.6.2.6] /* exists */ and certificate leaf[field.1.2.840.113635.100.6.1.13] /* exists */ and certificate leaf[subject.OU] = EQHXZ8M8AV
```

Reply:
Ah I didn't get that this was a mini language that you could use to express the check...

Reply:
Are these properties not returned as part of Context ?

Reply:
I still suspect if we extract those properties out it might be more useful. For client posture check it might be okay to have the requirement being passed into verify function but usually I feel exporting these properties and analytics done later might be a more realistic scenario.

Reply:
I don't think the macOS APIs expose all that data.  We do export the ones we get (team ID, etc.), but not all the fields in the cert or chain.